### PR TITLE
ref(staff): Allow helper to activate staff and update permission tests

### PR DESCRIPTION
**Pull request by schew2381**
`make_request` should be updated to handle activating staff. Also remove redundant helpers in permission tests

### CODE REVIEW

File: tests/sentry/api/test_permissions.py [hunk 1]
```diff
@@ -5,51 +5,38 @@
...
-
-    def _activate_superuser(self):
-        self.request.superuser.uid = str(self.user.id)
-        self.request.superuser._is_active = True
-
-    def _activate_staff(self):
-        self.request.staff.uid = str(self.user.id)
-        self.request.staff._is_active = True
+        self.superuser_request: Request = self.make_request(user=self.user, is_superuser=True)  # type: ignore
+        self.staff_request: Request = self.make_request(user=self.user, is_staff=True)  # type: ignore
```
Comment (cathteng) [inline tests/sentry/api/test_permissions.py:15]: why are we ignoring the type of this again?

**Review by cathteng — APPROVED** [review]

File: tests/sentry/api/test_permissions.py (hunk 1 above)
Comment (schew2381) [reply tests/sentry/api/test_permissions.py:15]: the helper returns httprequest but we expect drf request in the permission checks

Comment (cathteng) [reply tests/sentry/api/test_permissions.py:15]: i see. could you cast `HttpRequest` into `Request`?

Comment (schew2381) [reply tests/sentry/api/test_permissions.py:15]: I tried that but it was overriding something in the helper

**Pull request merged**